### PR TITLE
Convert CookieBanner to an island

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

Whilst working on the new CMP (consent management platform) I noticed the current CMP (aka the "Cookie Banner") isn't showing on DCR pages.

The "Cookie Banner" component should be displayed to any users viewing the Guardian who haven't granted consent (users who haven't set the `GU_TK` cookie). 

Currently there's a `<CookieBanner>` component, but the code for it isn't running in the client because it hasn't been converted to an "island". Therefore user's viewing a DCR page never see the banner, even if they should. This PR fixes that.

## Why?

Get user's consent state for third party tracking and personalised ads.

## Screenshot (mobile)

cmp-dcr

### CODE REVIEW

File: src/web/browser/cookie.ts [hunk 1]
```diff
@@ -6,6 +6,11 @@ const isValidCookieValue = (name: string): boolean =>
 
 const getShortDomain = (isCrossSubdomain = false): string => {
     const domain = document.domain || '';
+
+    if (domain === 'localhost') {
```
Comment (GHaberis) [inline src/web/browser/cookie.ts]: This addition to `cookie.ts` fixes a bug where the consent Cookie was never successfully set on `localhost`, the same issue was recently(ish) fixed on `frontend`: https://github.com/guardian/frontend/pull/21650

File: src/web/islands/islands.tsx [hunk 2]
```diff
@@ -124,6 +125,11 @@ export const hydrateIslands = (
             },
             root: 'reader-revenue-links-footer',
         },
+        {
+            component: CookieBanner,
+            props: {},
+            root: 'cookie-banner',
+        },
```
Comment (oliverlloyd) [inline src/web/islands/islands.tsx]: This is the perfect placement, thanks for thinking about this!

File: src/web/browser/cookie.ts (hunk 1 above)
Comment (oliverlloyd) [reply src/web/browser/cookie.ts]: I tested this locally and yep, it now dismisses correctly

File: src/web/layouts/StandardLayout.tsx [hunk 3]
```diff
@@ -161,7 +160,7 @@ export const StandardLayout = ({ CAPI, config, NAV }: Props) => {
                 />
             </Section>
 
-            <CookieBanner />
+            <div data-island="cookie-banner" />
```
Comment (oliverlloyd) [inline src/web/layouts/StandardLayout.tsx]: I just went ahead and added a commit to change this to this simpler structure

Comment (GHaberis) [reply src/web/layouts/StandardLayout.tsx]: @oliverlloyd thanks for the review and correcting the use of a `<Section>` for this Island!

**Pull request merged**